File: automacao/aula2cucumber/src/test/java/br/com/aula/steps/LoginSteps.java
package br.com.aula.steps;

import br.com.aula.pages.DashboardPage;
import br.com.aula.pages.HomePage;
import br.com.aula.pages.LoginPage;
import cucumber.api.java.it.E;
import cucumber.api.java.it.Quando;
import cucumber.api.java.pt.Dado;
import cucumber.api.java.pt.Entao;
import io.qameta.allure.Description;
import io.qameta.allure.Epic;
import io.qameta.allure.junit4.DisplayName;
import org.junit.Assert;
import org.junit.Test;

public class LoginSteps {

    LoginPage loginPage = new LoginPage();


    @E("que preencho o username valido")
    public void escreverNoCampoLogin() {
        loginPage.escreverNoCompoLogin();
    }

    @E("que preencho a senha valida")
    public void escreverNoCampoSenha() {
        loginPage.escreverNoCompoSenha();
    }

    @Quando("clico em Login")
    public void clicarNoBotaoLogin() {
        loginPage.clicarNoBotaoLogin();
    }
    @Entao("devo vizualizar uma mensagem de boas vindas")
    public void acessarTelaLogin() {
        loginPage.clicarNoBotaoLogin();
    }




//    @Test
//    @Epic("Login")
//    @DisplayName("validar Login Com Username E Senha Valida")
//    public void validarLoginComUsernameESenhaValidos() {
//        loginPage.escreverNoCompoLogin();
//        loginPage.escreverNoCompoSenha();
//        loginPage.clicarNoBotaoLogin();
//
//        Assert.assertEquals("Welcome teste123", dashboardPage.validarTextLogadoNaTela());
//    }

}

File: automacao/aula3cypresswebCucumber/cypress/integration/steps/LoginSteps.js
/// <reference types="cypress" />

import LoginPage from "../pages/LoginBase"
const loginPage = new LoginPage;

Given("que preencho o username valido", ()=>{
    loginPage.preencherUsername();
  });
  
And("que preencho a senha valida", ()=>{
    loginPage.preencherSenha();
  });
  
When("clico em login", ()=>{
    loginPage.clicarNoBtnLogin();
  });
  
Given("que preencho o username invalido", ()=>{
    loginPage.preencherUsernameErro();
  });
  
And("que preencho a senha invalida", ()=>{
    loginPage.preencherSenhaErro();
  });
  
Then("devo visualizar uma mensagem de erro", ()=>{
    loginPage.validarMsgErro();
})

